src/Pate/Monad.hs

```diff
@@ -380,126 +388,96 @@ freshSimVars blocks = do
   regs <- MM.mkRegStateM (\r -> unconstrainedRegister argNames r)
   return $ SimBoundVars regs (SimState mem (MM.mapRegsWith (\_ -> PSR.macawVarEntry) regs))
 
-
+-- Although 'AssumptionSet' has a scope parameter, the current interface doesn't have a
+-- good mechanism for enforcing the fact that we are only pushing assumptions that
+-- are actually scoped properly.
+-- This is manifest in the fact that the background frame in 'envCurrentFrame' doesn't
+-- track any scope, and is therefore unsafely coerced into any target scope.
+-- TODO: Rather than trying to enforce this statically (which would be difficult and require
```

Comment:
This seems important to not lose track of - can we add a github issue that we can reference from the code here?